### PR TITLE
PM-12734: Add set up autofill action card

### PR DESCRIPTION
**Pull request by matt-livefront**
## 🎟️ Tracking

[PM-12734](https://bitwarden.atlassian.net/browse/PM-12734) - Add autofill action card
[PM-12735](https://bitwarden.atlassian.net/browse/PM-12735) - Display turn on autofill screen

## 📔 Objective

Adds the set up autofill action card to the Settings > Autofill screen and navigation to the turn on autofill screen.

## 📸 Screenshots

Screenshot 2024-10-02 at 11 24 20 AM

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[PM-12734]: https://bitwarden.atlassian.net/browse/PM-12734?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PM-12735]: https://bitwarden.atlassian.net/browse/PM-12735?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by fedemkr — DISMISSED** [review]

File: BitwardenShared/UI/Platform/Settings/Settings/AutoFill/AutoFillView.swift [hunk 1]
```diff
@@ -14,19 +14,42 @@ struct AutoFillView: View {
 
     var body: some View {
         VStack(spacing: 20) {
+            autofillActionCard
+
             autoFillSection
 
             additionalOptionsSection
         }
+        .animation(.easeInOut, value: store.state.badgeState?.autofillSetupProgress == .complete)
```
Comment (phil-livefront) [inline BitwardenShared/UI/Platform/Settings/Settings/AutoFill/AutoFillView.swift:23]: super small nit, but should this logic here be done as a computed property in the state?

File: BitwardenShared/UI/Platform/Settings/Settings/AutoFill/AutoFillView.swift [hunk 2]
```diff
@@ -14,19 +14,42 @@ struct AutoFillView: View {
...
+        .task {
+            await store.perform(.streamSettingsBadge)
+        }
     }
 
     // MARK: Private views
 
+    /// The action card for setting up autofill.
+    @ViewBuilder private var autofillActionCard: some View {
+        if let progress = store.state.badgeState?.autofillSetupProgress, progress != .complete {
```
Comment (phil-livefront) [inline BitwardenShared/UI/Platform/Settings/Settings/AutoFill/AutoFillView.swift:38]: same comment here, just to move logic out of the view.  feel free to ignore tho.

**Review by phil-livefront — APPROVED** [review]

File: BitwardenShared/UI/Platform/Settings/Settings/AutoFill/AutoFillView.swift (hunk 1 above)
Comment (matt-livefront) [reply BitwardenShared/UI/Platform/Settings/Settings/AutoFill/AutoFillView.swift:23]: This is pretty simple, is there advantage of moving it to the state? This vs exposing a `isAutofillSetupProgressComplete` variable would be similar?

Comment (phil-livefront) [reply BitwardenShared/UI/Platform/Settings/Settings/AutoFill/AutoFillView.swift:23]: no real advantage and i totally agree is pretty simple.  I guess I was looking to far into the pattern of removing any business logic from the view so its "dumb"

Comment (matt-livefront) [reply BitwardenShared/UI/Platform/Settings/Settings/AutoFill/AutoFillView.swift:23]: Makes sense, I agree with that in general though!

**Review by phil-livefront — APPROVED** [review]

**Review by fedemkr — APPROVED** [review]

**Pull request merged**